src/prediction/predict_race.py

```diff
@@ -0,0 +1,36 @@
+from PIL import Image, UnidentifiedImageError
+import requests
+import numpy as np
+from deepface import DeepFace
+from torch.utils.data import Dataset
+from typing import Optional
+class RacePredictor():
+    """A race predictor for images with faces."""
+
+    def __init__(self,
+                 dataset: Optional[Dataset]):
```

Comment:
Let's not do the heavy logic in the `__init__`. This is equivalent of a constructor in other OOP languages.

If you want, you can override the `__call__` method (look it up on Google)